setools/infoflow.py

```diff
@@ -344,6 +348,34 @@ def _build_subgraph(self):
 
             self.subG.remove_edges_from(delete_list)
 
+        if self.conditionals is not None:
+            delete_list = []
+            for s, t in self.subG.edges():
+                edge = Edge(self.subG, s, t)
+                rule_list = []
+                for rule in iter(edge.rules):
+                    try:
+                        if rule.conditional:
+                            bool_enabled = rule.conditional.evaluate(**self.conditionals)
+                            # if conditional is true then delete the false rules
+                            if bool_enabled:
+                                for false_rule in rule.conditional.false_rules():
+                                    rule_list.append(false_rule)
+                            # if conditional is false then delete the true rules
+                            else:
+                                for true_rule in rule.conditional.true_rules():
+                                    rule_list.append(true_rule)
+                    except RuleNotConditional as e:
+                        pass
+                for rule in rule_list:
+                    try:
+                        edge.rules.remove(rule)
+                    except ValueError:
```

Comment:
In what case is there a `ValueError`?

Reply:
There are two cases which I broke out into ifs to avoid the exception. In one case calls to false_rules and true_rules will return rules that are not applicable to the current edge. In the other case true_rules or false_rules will return the same rule multiple times and it can only be removed once.